### PR TITLE
Analysis iter config pure dataclass

### LINKED ISSUE #4068: Make AnalysisIterConfig a proper dataclass
AnalysisIterConfig is a `dataclass` with setter and getter logic, that is not really needed. 

Setting and getting logic should be lifted up into the AnalysisConfig object

### PR DESCRIPTION
**Pull request by DanSava**
**Issue**
Resolves #4068

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

**Comment by DanSava** [comment]
Jenkins test this, please!

File: src/ert/_c_wrappers/enkf/analysis_config.py [hunk 1]
```diff
@@ -210,6 +207,27 @@ def minimum_required_realizations(self) -> int:
     def have_enough_realisations(self, realizations) -> bool:
         return realizations >= self.minimum_required_realizations
 
+    @property
+    def case_format(self) -> Optional[str]:
```
Comment (eivindjahren) [inline src/ert/_c_wrappers/enkf/analysis_config.py:211]: There seems to be a problem annotated in the initialization:
doing `analysis_iter_config=AnalysisIterConfig()` makes just one `AnalsysIterConfig` for all defaulted `AnalysisConfig`. need to use:

```python
__init__(
analysis_iter_config=None,
....
):
  if analysis_iter_config is None:
      analysis_iter_config = AnalysisIterConfig()
```

**Review by eivindjahren — APPROVED** [review]
There is an error in the `__init__` of AnalysisConfig, not related to your PR that needs to be fixed, but we can merge this now as long as we get that fixed.

File: src/ert/_c_wrappers/enkf/analysis_config.py (hunk 1 above)
Comment (DanSava) [reply src/ert/_c_wrappers/enkf/analysis_config.py:211]: Very good catch! I will include the fix as a separate commit to this pr

**Pull request merged**